core/backend/src/NativeHost.ts

```diff
@@ -198,7 +201,7 @@ export class NativeHost {
   public static get authorization() { return IModelHost.authorizationClient as NativeAppAuthorizationBackend; }
 
   /** Event called when the user's sign-in state changes - this may be due to calls to signIn(), signOut() or because the token was refreshed */
-  public static readonly onUserStateChanged = new BeEvent<(token?: AccessToken) => void>();
+  public static readonly onUserStateChanged = new BeEvent<(token?: AccessTokenString) => void>();

```

Comment:
redundant typing

Reply:
Why are we defining `AccessTokenString = string | undefined` when none of our other string-y types like GuidString and Id64String include `undefined` in their typing?

Wouldn't it be more useful to drop the undefined and allow functions to accept `AccessTokenString | undefined` so there can be an expectation of an AccessToken actually existing instead of needing to check for `undefined` every time we pass an `AccessTokenString` around?

Reply:
@evelynpreslar-bentley In making that change, there's several methods that return just `AccessTokenString` but their member variable `_accessToken` is of type `AccessTokenString?`, so we would have to add some kind of default response. Would we want just an empty string if it is undefined?

Reply:
If a function needs to return a default response, then the return type should be `AccessTokenString | undefined`, no?
That way, undefined can be returned by certain function without polluting the `AccessTokenString` type definition.
Empty strings are values (invalid ones, in this case), but values nonetheless, because they're expressly **not** `undefined`

If we bake the undefined into the type definition, then anything that deals with AccessTokenString needs to check whether it's undefined before doing anything with it. We expect `GuidString` to not be an empty string or undefined when we receive one, why not `AccessTokenString`?

For functions like [`getAccessToken`](https://github.com/imodeljs/imodeljs/blob/bc1628448022ba467a5a62ceb7358c5e9f2b678d/clients/itwin/src/AuthorizationClient.ts#L16-L17), we previously expected an `AccessToken` object as a return value in the Promise. Now, the value is effectively `string | undefined`, which isn't 1-to-1 because the previous typing wasn't `AccessToken | undefined`

Reply:
@evelynpreslar-bentley I've updated `AccessTokenString = string` and updated all uses of it to take in `AccessTokenString | undefined` for the places that needed it. From the way I understand the preexisting code (correct me if wrong), if the accessToken object was passed `undefined` for the tokenString then it replaced it with an empty string which means even if an AccessToken was always be returned from a function it's `AccessToken.toTokenString()` method could still return an empty string. Not sure what the best 1-to-1 equivalent would be without actually returning an empty string.

Reply:
An `AccessToken` constructed with an empty string is simply an invalid access token.
The token object is still defined, it's just invalid and shouldn't be usable for anything that requires one.
For code that doesn't care about what's in the token though (and simply passes it along) we only care if something fails _because_ of the bad token, not the reason why it was bad (i.e. empty string, token header missing, etc.)

An `AccessToken` constructed with an empty string before is directly comparable to `tokenString = ""`.
Both are not `undefined` and both should fail in the same manner when passed to something expecting a valid token, so there shouldn't be any other mapping issues to worry about between the two typings.

Reply:
Closing this as resolved with the latest code changes.